lib/runtime-c-api/src/lib.rs

```diff
@@ -569,6 +574,99 @@ pub unsafe extern "C" fn wasmer_export_descriptor_kind(
     named_export_descriptor.kind.clone()
 }
 
+/// Serialize the given Module.
+///
+/// It's up to the caller to free the memory of the
+/// `serialized_module` byte array (both the `bytes` field and the
+/// structure).
+///
+/// Returns `wasmer_result_t::WASMER_OK` upon success.
+///
+/// Returns `wasmer_result_t::WASMER_ERROR` upon failure. Use `wasmer_last_error_length`
+/// and `wasmer_last_error_message` to get an error message.
+#[allow(clippy::cast_ptr_alignment)]
+#[no_mangle]
+pub unsafe extern "C" fn wasmer_module_serialize(
+    serialized_module: *mut *mut wasmer_byte_array,
+    module: *const wasmer_module_t,
+) -> wasmer_result_t {
+    let module = &*(module as *const Module);
+
+    match module.cache() {
+        Ok(artifact) => match artifact.serialize() {
+            Ok(serialized_artifact) => {
+                *serialized_module = Box::into_raw(Box::new(wasmer_byte_array {
+                    bytes: serialized_artifact.as_ptr(),
+                    bytes_len: serialized_artifact.len() as u32,
+                })) as *mut wasmer_byte_array;
+
+                mem::forget(serialized_artifact);
```

Comment:
Does this leak memory?

Reply:
Yes. If we don't, then the `wasmer_byte_array.bytes` field will contain a null pointer. It's up to the caller to free the memory.

Reply:
@Hywan how about a `wasmer_serialized_module_t *` that points to the `serialized_artifact` on the rust side and a function to access the bytes `wasmer_byte_array wasmer_serialized_module_bytes(wasmer_module_t *serialized_module)`?  This would be similar to this: https://github.com/Hywan/wasmer/blob/f049e9b0b378ae52b6cef467364f25fcff5ec2d8/lib/runtime-c-api/src/lib.rs#L557

Reply:
That's a good idea!

Reply:
Done.